src/file.cr

```diff
@@ -237,25 +240,83 @@ class File < IO::FileDescriptor
     basename(path).chomp(suffix)
   end
 
+  private alias User = System::User | String | UInt32 | Int32
+  private alias Group = System::Group | String | UInt32 | Int32
+
   # Changes the owner of the specified file.
   #
   # ```
-  # File.chown("/foo/bar/baz.cr", 1001, 100)
-  # File.chown("/foo/bar", gid: 100)
+  # File.chown("/foo/bar/baz.cr", "a_owner", "a_group")
+  # File.chown("/foo/bar", group: "a_group")
```

Comment:
Minor: maybe use a descriptive example user/group name such as `alice` and `admin` instead of `a_user` and `a_group`? Same below.